src/ir_Argo.cpp

```diff
@@ -0,0 +1,282 @@
+/*
+Node MCU/ESP8266 Sketch to emulate Argo Ulisse 13 DCI remote
+Copyright 2017 Schmolders
+*/
+
+#include "ir_Argo.h"
+#include <algorithm>
+#include "IRremoteESP8266.h"
+#include "IRutils.h"
+#include <Arduino.h>
+
+
+//  ARgo
+
+// Constants
+// using SPACE modulation. MARK is always const 400u
+#define ARGO_PREAMBLE_1           6400U  // Mark
+#define ARGO_PREAMBLE_2           3300U  // Space
+#define ARGO_MARK                  400U
+#define ARGO_ONE_SPACE            2200U
+#define ARGO_ZERO_SPACE            900U
+
+#if SEND_ARGO
+// Send a Argo A/C message.
+//
+// Args:
+//   data: An array of ARGO_COMMAND_LENGTH bytes containing the IR command.
+//
+// Status: STABLE
+//
+// Overloading the IRSend Function
+
+void IRsend::sendArgo(unsigned char data[], uint16_t nbytes,
+                        uint16_t repeat) {
+  if (nbytes < ARGO_COMMAND_LENGTH)
+    return;  // Not enough bytes to send a proper message.
+  // Set IR carrier frequency
+  enableIROut(38);
+  for (uint16_t r = 0; r <= repeat; r++) {
+    // Header
+    // TODO(kaschmo): validate
+    mark(ARGO_PREAMBLE_1);
+    space(ARGO_PREAMBLE_2);
+    // send data, defined in IRSend.cpp
+    for (uint16_t i = 0; i < nbytes; i++)
+      sendData(ARGO_MARK, ARGO_ONE_SPACE, ARGO_MARK,
+               ARGO_ZERO_SPACE, data[i], 8, false);
+               // send LSB first reverses the bit order in array for sending.
+  }
+}
+
+IRArgoESP::IRArgoESP(uint16_t pin) : _irsend(pin) {
+  stateReset();
+}
+
+void IRArgoESP::begin() {
+  _irsend.begin();
+}
+
+void IRArgoESP::send() {
+  Serial.println("Sending IR code");
```

Comment:
Please remove. We shouldn't be printing to serial from the library, except as a debug command.
e.g. DPRINTLN()